src/acachecontrol/cache.py

```diff
@@ -18,18 +19,33 @@ def __init__(self):
     def __contains__(self, key):
         return self._make_key_hashable(key) in self.cache
 
-    def add(self, key: Tuple[str, str, Dict], value: Any) -> None:
+    def add(self, key: Tuple[str, str, Dict], value: Any, headers: Any):
         """Add value to the cache.
 
+        headers - any dict-like obj
+
         # TODO: Values should be stored within the following structure:
         {'created_at': <unix_timestamp>, 'max_age': <seconds>, 'value': <value>}
         """
+        cc_header = self.parse_cache_control_header(headers)
         hashable_key = self._make_key_hashable(key)
-        self.cache[hashable_key] = value
+        self.cache[hashable_key] = {
+            "created_at": time.time(),
+            "max-age": cc_header.get("max-age"),
+            "value": value,
+        }
         self.release_new_key(key)
 
     def get(self, key: Tuple[str, str, Dict]) -> Any:
-        return self.cache.get(self._make_key_hashable(key))
+        try:
+            cache_entry = self.cache.get(self._make_key_hashable(key))
+            if cache_entry:
+                return cache_entry.get("value")
```

Comment:
```suggestion
                return cache_entry["value"]
```
Is `value` a required key?